packages/components/src/mobile/link-picker/index.native.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { SafeAreaView } from 'react-native';
+import { startsWith } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { BottomSheet } from '@wordpress/components';
+import { getProtocol, prependHTTP } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import LinkPickerResults from './link-picker-results';
+import NavigationHeader from '../bottom-sheet/navigation-header';
+import styles from './styles';
+
+// this creates a search suggestion for adding a url directly
+export const createDirectEntry = ( value ) => {
+	let type = 'URL';
+
+	const protocol = getProtocol( value ) || '';
+
+	if ( protocol.includes( 'mailto' ) ) {
+		type = 'mailto';
+	}
+
+	if ( protocol.includes( 'tel' ) ) {
+		type = 'tel';
+	}
```

Comment:
I'm not sure if there is such case but the above checks can be enhanced to be case insensitive (e.g. the link may be MAILTO or Mailto)

Reply:
Good idea Antonis! I've updated to make the protocol detection case-insensitive. :+1: